presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/LimitPushDown.java

```diff
@@ -94,6 +99,15 @@ public PlanNode visitPlan(PlanNode node, RewriteContext<LimitContext> context)
         @Override
         public PlanNode visitLimit(LimitNode node, RewriteContext<LimitContext> context)
         {
+            // replace nodes with empty ValuesNode in case of limit 0
+            if (node.getCount() == 0) {
+                List<Expression> values = node.getOutputSymbols().stream().map(s -> new NullLiteral()).collect(toList());
+                ValuesNode valuesNode = new ValuesNode(idAllocator.getNextId(),
+                                                        node.getOutputSymbols(),
+                                                        ImmutableList.of(values));
+                return new LimitNode(idAllocator.getNextId(), valuesNode, node.getCount());
```

Comment:
This might work with just a Values node with 0 rows, but can't be sure without trying it out.